qiita_ware/ebi.py

```diff
@@ -455,20 +376,20 @@ def generate_experiment_xml(self):
 
         Returns
         -------
-        xml.etree.Element
-            The root elelement of the generated ``ElementTree``
+        str
+            string with experiment XML values
         """
         study_alias = self._get_study_alias()
         experiment_set = ET.Element('EXPERIMENT_SET', {
             "xmlns:xsi": "http://www.w3.org/2001/XMLSchema-instance",
             "xsi:noNamespaceSchemaLocation": "ftp://ftp.sra.ebi.ac.uk/meta/xsd"
                                              "/sra_1_3/SRA.experiment.xsd"})
-        for sample_name, sample_info in sorted(self.samples.items()):
+        for sample_name, sample_prep in sorted(self.samples_prep.items()):
             sample_alias = self._get_sample_alias(sample_name)
-
             experiment_alias = self._get_experiment_alias(sample_name)
+            sample_prep = dict(self.samples_prep[sample_name])
```

Comment:
Well, those are created as objects in qiita_db so converting to dicts for code simplicity